tests/php/test-amp-style-sanitizer.php

```diff
@@ -945,6 +945,38 @@ static function( $preempt, $request, $url ) {
 		}
 	}
 
+	/**
+	 * Test get_stylesheet_from_url with a bad URL.
+	 *
+	 * @covers AMP_Style_Sanitizer::get_stylesheet_from_url()
+	 */
+	public function test_get_stylesheet_from_url_bad_url() {
+		$dom = Document::fromHtml( '<html><head></head><body></body></html>', Options::DEFAULTS );
+
+		$sanitizer = new AMP_Style_Sanitizer( $dom, [] );
+
+		$css_url    = 'https://example.com/style.css';
+		$stylesheet = $this->call_private_method( $sanitizer, 'get_stylesheet_from_url', [ $css_url ] );
+
+		$this->assertTrue( is_wp_error( $stylesheet ) );
+		$this->assertInstanceOf( WP_Error::class, $stylesheet );
+		$this->assertStringStartsWith( 'Failed to fetch:', $stylesheet->get_error_message() );
+
+		$css_url    = amp_get_asset_url( 'css/amp-default.css' );
+		$stylesheet = $this->call_private_method( $sanitizer, 'get_stylesheet_from_url', [ $css_url ] );
+
+		$this->assertIsString( $stylesheet );
+		$this->assertNotEmpty( $stylesheet );
+		$this->assertFalse( is_wp_error( $stylesheet ) );
+
+		$css_url    = '/wp-includes/css/admin-bar.css';
+		$stylesheet = $this->call_private_method( $sanitizer, 'get_stylesheet_from_url', [ $css_url ] );
```

Comment:
Are these triggering external HTTP requests? If so, we should add a `pre_http_request` filter as you can see elsewhere among the tests in this class to do mocking to prevent the overhead of a request.

Reply:
Added the filter 👍🏼